src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.DocumentChangeTracker.cs

```diff
@@ -81,13 +87,25 @@ public void StartTracking(Uri documentUri, SourceText initialText)
                 Contract.ThrowIfTrue(_trackedDocuments.ContainsKey(documentUri), $"didOpen received for {documentUri} which is already open.");
 
                 _trackedDocuments.Add(documentUri, initialText);
+
+                // If we can't find the document in any of the registered workspaces, add it to our loose files workspace.
+                if (!IsPresentInRegisteredWorkspaces(documentUri, _lspWorkspaceRegistrationService))
+                {
+                    _lspMiscellaneousFilesWorkspace.AddMiscellaneousDocument(documentUri);
+                }
             }
 
             public void UpdateTrackedDocument(Uri documentUri, SourceText text)
             {
                 Contract.ThrowIfFalse(_trackedDocuments.ContainsKey(documentUri), $"didChange received for {documentUri} which is not open.");
 
                 _trackedDocuments[documentUri] = text;
+
+                // If we see the document has been moved to a registered workspace, remove it from our loose files workspace.
+                if (IsPresentInRegisteredWorkspaces(documentUri, _lspWorkspaceRegistrationService))
```

Comment:
This won't be expensive, right?

Reply:
Nope, we do this on every request already.  It just looks for docs by file path which ends up just looking at a map in the solution state.